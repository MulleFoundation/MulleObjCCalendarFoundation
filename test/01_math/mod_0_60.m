#include <stdio.h>
#include <assert.h>


static int  mod_0_n( int *p_value, int max)
{
   int   diff;
   int   value;

   diff  = 0;
   value = *p_value;

   if( value < 0)
   {
      diff   = value / max - 1;
      value %= max;
      if( value)
         value += max;
   }
   else
      if( value >= max)
      {
         diff   = value / max;
         value %= max;
      }

   *p_value = value;
   return( diff);
}


int  main( void)
{
   int   i;
   int   value;
   int   diff;

   for( i = -125; i <= 125; i +=5)
   {
      value = i;
      diff  = mod_0_n( &value, 60);
      printf( "%d: mod=%d diff=%d\n", i, value, diff);
   }
   return( 0);
}

