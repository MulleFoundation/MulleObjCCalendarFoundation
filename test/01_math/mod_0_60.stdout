-125: mod=55 diff=-3
-120: mod=0 diff=-3
-115: mod=5 diff=-2
-110: mod=10 diff=-2
-105: mod=15 diff=-2
-100: mod=20 diff=-2
-95: mod=25 diff=-2
-90: mod=30 diff=-2
-85: mod=35 diff=-2
-80: mod=40 diff=-2
-75: mod=45 diff=-2
-70: mod=50 diff=-2
-65: mod=55 diff=-2
-60: mod=0 diff=-2
-55: mod=5 diff=-1
-50: mod=10 diff=-1
-45: mod=15 diff=-1
-40: mod=20 diff=-1
-35: mod=25 diff=-1
-30: mod=30 diff=-1
-25: mod=35 diff=-1
-20: mod=40 diff=-1
-15: mod=45 diff=-1
-10: mod=50 diff=-1
-5: mod=55 diff=-1
0: mod=0 diff=0
5: mod=5 diff=0
10: mod=10 diff=0
15: mod=15 diff=0
20: mod=20 diff=0
25: mod=25 diff=0
30: mod=30 diff=0
35: mod=35 diff=0
40: mod=40 diff=0
45: mod=45 diff=0
50: mod=50 diff=0
55: mod=55 diff=0
60: mod=0 diff=1
65: mod=5 diff=1
70: mod=10 diff=1
75: mod=15 diff=1
80: mod=20 diff=1
85: mod=25 diff=1
90: mod=30 diff=1
95: mod=35 diff=1
100: mod=40 diff=1
105: mod=45 diff=1
110: mod=50 diff=1
115: mod=55 diff=1
120: mod=0 diff=2
125: mod=5 diff=2
